picard/formats/vorbis.py

```diff
@@ -231,12 +231,22 @@ def _save(self, filename, metadata):
         if file.tags is None:
             file.add_tags()
         if config.setting["clear_existing_tags"]:
-            channel_mask = file.tags.get('waveformatextensible_channel_mask', None)
+            preserve_tags = ['waveformatextensible_channel_mask']
+            if not is_flac and config.setting["preserve_images"]:
+                preserve_tags.append('METADATA_BLOCK_PICTURE')
+                preserve_tags.append('COVERART')
+            preserved_values = {}
+            for name in preserve_tags:
+                if name in file.tags:
+                    preserved_values[name] = file.tags[name]
             file.tags.clear()
-            if channel_mask:
-                file.tags['waveformatextensible_channel_mask'] = channel_mask
+            for name, value in preserved_values.items():
+                if value:
```

Comment:
Is `if value:` really needed?

Reply:
Yes, because above the tags get added unconditionally to the `preserved_values` dict, and if it was empty we would add an empty tag. The alternative would be to do it the other way around and only actually put those tags to the `preserved_values` dict if they are set, but I found the code to read nicer this way. Otherwise it doesn't matter much really.

Reply:
I switched it around